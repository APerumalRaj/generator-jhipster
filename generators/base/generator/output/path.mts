import normalize from 'normalize-path';

export function normalizeOutputPath(outputPath: any): any {
  return outputPath ? normalize(outputPath) : outputPath;
}

export function applyPathCustomizer(context: any, outputPath: any, outputPathCustomizer: any): any {
  if (Array.isArray(outputPathCustomizer)) {
    let outputhPathApplied = outputPath;
    outputPathCustomizer.forEach(customizer => {
      outputhPathApplied = customizer.call(context, outputhPathApplied);
    });
    return outputhPathApplied;
  }
  return outputPathCustomizer.call(context, outputPath);
}
