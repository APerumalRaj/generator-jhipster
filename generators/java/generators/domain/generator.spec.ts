import { basename, dirname, resolve } from 'path';
import { fileURLToPath } from 'url';
import { before, it, describe, expect } from 'esmocha';

import Generator from './index.js';
import { shouldSupportFeatures, testBlueprintSupport } from '../../../../test/support/tests.js';
import { defaultHelpers as helpers, result } from '../../../../testing/index.js';

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

const generator = `${basename(resolve(__dirname, '../../'))}:${basename(__dirname)}`;

describe(`generator - ${generator}`, () => {
  shouldSupportFeatures(Generator);
  describe('blueprint support', () => testBlueprintSupport(generator));

  describe('with defaults options', () => {
    before(async () => {
      await helpers.runJHipster(generator).withJHipsterConfig();
    });

    it('should match files snapshot', () => {
      expect(result.getStateSnapshot()).toMatchSnapshot();
    });
  });
});
