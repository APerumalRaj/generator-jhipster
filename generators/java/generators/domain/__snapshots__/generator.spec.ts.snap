// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generator - java:domain with defaults options should match files snapshot 1`] = `
{
  ".yo-rc.json": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/domain/AssertUtils.java": {
    "stateCleared": "modified",
  },
}
`;
